chat-plugins/panagrams-beta.js

```diff
@@ -152,4 +152,108 @@ exports.commands = {
+				if (toId(guess.species) === toId(Game.answer.species)) {
+					this.add('|html|<b>' + user.name + '</b> guessed <b>' + guess.species + '</b>, which was the correct answer! Congratulations!');
+					Game.end();
+				} else if (guess.num < 1 || guess.baseSpecies) user.sendTo(this, guess.species + ' is either an alternate form or doesn\'t exist in the games. These cannot be guessed.');
+				else {
+					if (Object.keys(Game.users).length < 3) {
+						this.add('|html|<b>' + user.name + '</b> guessed <b>' + guess.species + '</b>, but was not the correct answer...');
+						Game.guessed[toId(guess.species)] = user.userid;
+					} else {
+						this.add('|html|<b>' + user.name + '</b> guessed <b>' + guess.species + '</b>, but was not the correct answer. You have failed to guess the Pokemon, which was <b>' + (answer.name || answer.species) + '</b>');
+						Game.end();
+					}
+				}
+			} else {
+				if (message && message !== true) {
+					this.add('|c|' + user.getIdentity(this.id) + '|' + message);
+				}
+			}
+			this.update();
+		};
+	}
+	Panagram.prototype.end = function (forced) {
+		if (forced) this.room.add('The game of panagram has been forcibly ended.');
+		this.room.chat = Rooms.Room.prototype.chat;
+		if (this.sessions > 1 && !forced) pGames[this.room.id] = new Panagram(this.room, --this.sessions);
+		else delete pGames[this.room.id];
+	};
+	return Panagram;
+})();
+
+exports.commands = {
+	panagramrules: 'panagramhelp',
+	phelp: 'panagramhelp',
+	panagramhelp: function (target, room, user) {
+		if (!this.canBroadcast()) return;
+		this.sendReplyBox('<b>Panagram Commands</b><br>' +
+			'<li>/panagram <em>Category</em> - Starts a game of panagram that follows the specified category. Valid categories include <i>Pokemon, Items, Abilities</i> and <i>moves</i>. Not specifying a category starts a pokemon panagram by default. Requires + or higher to use.' +
+			'<li>/panagram end <small>or</small> /endp - Ends a game of panagram. Requires + or higher to use.<br>' +
+			'Users can guess answers by simply typing them into the chat.'
+		);
+	},
+
+	panagram: function (target, room, user, connection, cmd) {
+		if (pGames[room.id]) return this.errorReply("There is currently a game of panagram going on in this room.");
+		if (!this.can('broadcast', null, room)) return this.errorReply("You must be ranked + or higher to start a game of panagram in this room.");
+		if (!target || isNaN(target)) target = 1;
+		if (target < 1) return this.errorReply('You cannot set the number of sessions to anything less than 1.');
+
+		pGames[room.id] = new Panagram(room, Number(target));
+	},
+
+	endp: 'panagramend',
+	panagramend: function (target, room, user) {
+		if (!pGames[room.id]) return this.errorReply("There is no game of panagram going on in this room.");
+		if (!this.can('broadcast', null, room)) return this.sendReply("You must be ranked + or higher to end a game of panagram in this room.");
```

Comment:
This can be voice, however...

It would be neat to make a command to skip the current panagram if everyone gets stuck.  Thoughts?